### PR TITLE
Bug 1989335: pkg/operator/targetconfigcontroller: block rollout and report on missing external resource

### PR DESCRIPTION
**Pull request by hexfusion**
The current code reports the invalid resource but does not block a new revision from being created. In the case that configmap openshift-config-managed/csr-controller-ca is not managed by cluster-kube-controller-manager-operator block if SNO. This will ensure we don't rollout and cause disruption before critical resource generation.

Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

### CODE REVIEW

File: pkg/operator/targetconfigcontroller/targetconfigcontroller.go [hunk 1]
```diff
@@ -114,8 +116,9 @@ func createTargetConfig(ctx context.Context, c TargetConfigController, recorder
 	if err != nil {
 		return false, err
 	}
-	if err := checkExternalDependencies(ctx, c.configMapLister, recorder); err != nil {
-		errors = append(errors, err)
+	if err := checkExternalDependencies(ctx, c.configMapLister, c.infrastructureLister, recorder); err != nil {
+		recorder.Warning("MissingRequiredResource", err.Error())
```
Comment (marun) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller.go:120]: Will `checkExternalDependencies` ever return an error that does not indicate a missing resource?

Comment (hexfusion) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: yeah, not in love with that.

Comment (hexfusion) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller.go]: updated.

**Review by lilic — COMMENTED** [review]
/lgtm

Note this still has a hold, CI failure is unrelated.

**Pull request merged**